registrations/test_serializers.py

```diff
@@ -0,0 +1,181 @@
+import datetime
+from django.test import TestCase
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+import pytz
+from rest_framework.serializers import ValidationError
+
+from registrations.serializers import (
+    MSISDNField, JembiAppRegistrationSerializer)
+
+
+def override_get_today():
+    return datetime.date(2016, 1, 1)
+
+
+class MSISDNFieldTests(TestCase):
+    def test_internal_value(self):
+        """
+        A phone number sent to the field should be formatted in E164 format
+        """
+        field = MSISDNField(country='ZA')
+        self.assertEqual(
+            field.run_validation('0821234567'), '+27821234567')
+
+    def test_bad_input(self):
+        """
+        If the input cannot be parsed, a validation error should be raised
+        """
+        field = MSISDNField()
+        self.assertRaisesMessage(
+            ValidationError,
+            'The string supplied did not seem to be a phone number',
+            field.run_validation, '1')
+
+    def test_impossible_phone_number(self):
+        """
+        If a phone number is not possible, a validation error should be raised
+        """
+        field = MSISDNField()
+        self.assertRaisesMessage(
+            ValidationError,
+            'Not a possible phone number',
+            field.run_validation, '+120012301')
+
+    def test_invalid_phone_number(self):
+        """
+        If a phone number is not valid, a validation error should be raised
+        """
+        field = MSISDNField()
+        self.assertRaisesMessage(
+            ValidationError,
+            'Not a valid phone number',
+            field.run_validation, '+12001230101')
+
+    def test_representation(self):
+        """
+        The representation should be in E164
+        """
+        field = MSISDNField(country='ZA')
+        self.assertEqual(
+            field.to_representation('0821234567'), '+27821234567')
+
+
+class JembiAppRegistrationSerializerTests(TestCase):
+    @mock.patch('ndoh_hub.utils.get_today', override_get_today)
+    def test_valid_registration(self):
+        """
+        If the registration is valid, then the serializer should be valid
+        """
+        self.maxDiff = None
+        data = {
+            'mom_msisdn': '0820000001',
+            'hcw_msisdn': '0820000002',
+            'mom_id_type': 'none',
+            'mom_dob': '1988-01-01',
+            'mom_lang': 'eng_ZA',
+            'mom_edd': '2016-06-06',
+            'mom_consent': True,
+            'clinic_code': '123456',
+            'mha': 1,
+            'created': '2016-01-01 00:00:00',
+        }
+        serializer = JembiAppRegistrationSerializer(data=data)
+        serializer.is_valid()
```

Comment:
Do we need this line if we call it on the next line again?

Reply:
No, I'll remove it